### PR TITLE
Feat: Move modules: `expected_failure` abort codes, replace block height with timestamp, refund assertions

### PR DESCRIPTION
**Pull request by andygolay**
# Summary
- Fix for issue #458 for specific error codes on expected failure
- Replace block height with timestamp in initiator module

# Changelog

In `atomic_bridge_initiator.move`, the following test functions got a specific abort code and location:

- ` ``   
    test_initiate_bridge_transfer_no_moveth
    ```
- ```    
    test_complete_bridge_transfer_wrong_preimage
   ```
- ```    
    test_refund_bridge_transfer
    ```
- ```    
    test_refund_completed_transfer
    ```

- In initiator module, block height is replaced with timestamp. 

- Time passing is simulated with `aptos_framework::timestamp::fast_forwa;rd_seconds(time_lock + 2)`

- `refund_bridge_transfer` on Move side and `refundBridgeTransfer` can only be called by originator  / owner.
    
# Testing

- Move unit tests pass.
- To verify, in `protocol-units/bridge/move-modules` run `movement move test`
- Foundry test may be failing currently

# Outstanding issues

- Seeing that we only have `expected_failure` tests in `atomic_bridge_initiator.move`, it would be good for `atomic_bridge_counterparty.move` to have some `expected_failure` tests.

### CODE REVIEW

File: protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move [hunk 1]
```diff
@@ -15,6 +15,8 @@ module atomic_bridge::atomic_bridge_counterparty {
     const COMPLETED: u8 = 2;
     const CANCELLED: u8 = 3;
 
+    const EINCORRECT_SIGNER: u64 = 7;
```
Comment (0xPrimata) [inline protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move:18]: feat branch has several error codes, errors are also not named, eincorrect_signer is now error 1. Might be good to get the changes from feat branch.

File: protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move [hunk 2]
```diff
@@ -159,12 +161,12 @@ module atomic_bridge::atomic_bridge_counterparty {
     }
     
     public fun abort_bridge_transfer(
-        caller: &signer,
+        account: &signer,
         bridge_transfer_id: vector<u8>
     ) acquires BridgeTransferStore, BridgeConfig {
         // check that the signer is the bridge_module_deployer
-        assert!(signer::address_of(caller) == borrow_global<BridgeConfig>(signer::address_of(caller)).bridge_module_deployer, 1);
```
Comment (0xPrimata) [inline protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move:166]: add named errors across the module

File: protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move [hunk 3]
```diff
@@ -448,8 +449,8 @@ module atomic_bridge::atomic_bridge_initiator {
         assert!(!aptos_std::smart_table::contains(&store.transfers, copy bridge_transfer_id), 300);
     }
 
-    #[test(creator = @moveth, aptos_framework = @0x1, sender = @0xdaff, atomic_bridge = @atomic_bridge)]
-    #[expected_failure]
+    #[test(creator = @moveth, aptos_framework = @0x1, sender = @origin_addr, atomic_bridge = @atomic_bridge)]
+    #[expected_failure ( abort_code = ENOT_INITIALIZED )]
```
Comment (andyjsbell) [inline protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move:453]: ditto

File: protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move [hunk 4]
```diff
@@ -364,7 +367,7 @@ module atomic_bridge::atomic_bridge_initiator {
     }
 
     #[test(creator = @moveth, aptos_framework = @0x1, sender = @0xdaff, atomic_bridge = @atomic_bridge)]
-    #[expected_failure]
+    #[expected_failure ( abort_code = EWRONG_PREIMAGE )]
```
Comment (andyjsbell) [inline protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move:370]: Nothing major but you can also tighten this constraint with 
`#[expected_failure(abort _code = EWRONG_PREIMAGE, location = Self)]`

Probably can drop the spacing here

File: protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move (hunk 1 above)
Comment (andygolay) [reply protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move]: Thanks; I've now adjusted the codes to be consistent with the ones in feature branch and use named errors throughout.

**Review by andyjsbell — APPROVED** [review]
lgtm

**Pull request merged**